workers/unity/Assets/Gdk/Core/Collections/BlittableList.cs

```diff
@@ -0,0 +1,52 @@
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Assets.Gdk.Core.Collections
+{
+    public struct BlittableList<T> : IList<T>
```

Comment:
I don't think these should implement IList or IDictionary as they aren't meant to be new collections; they are meant to be the existing collections served in an unusual way. They will act more like the collections they are trying to mimic if you do not provide conversions that will be preferentially picked over the intended ones.